src/sinks/azure_monitor_logs/config.rs

```diff
@@ -442,168 +253,21 @@ mod tests {
-        .unwrap();
-
-        let sink = AzureMonitorLogsSink::new(&config, None).unwrap();
-        let mut encoder = sink.build_encoder();
-
-        let mut log1 = [("message", "hello")].iter().copied().collect::<LogEvent>();
-        let (timestamp_key1, timestamp_value1) = insert_timestamp_kv(&mut log1);
-
-        let mut log2 = [("message", "world")].iter().copied().collect::<LogEvent>();
-        let (timestamp_key2, timestamp_value2) = insert_timestamp_kv(&mut log2);
-
-        let event1 = encoder.encode_event(Event::from(log1)).unwrap();
-        let event2 = encoder.encode_event(Event::from(log2)).unwrap();
-
-        let json1 = serde_json::to_string(&event1).unwrap();
-        let json2 = serde_json::to_string(&event2).unwrap();
-        let raw1 = RawValue::from_string(json1).unwrap();
-        let raw2 = RawValue::from_string(json2).unwrap();
-
-        let events = vec![raw1, raw2];
-
-        let request = sink.build_request_sync(events);
-
-        let (parts, body) = request.unwrap().into_parts();
-        assert_eq!(&parts.method.to_string(), "POST");
-
-        let json: serde_json::Value = serde_json::from_slice(&body[..]).unwrap();
-        let expected_json = serde_json::json!([
-            {
-                timestamp_key1: timestamp_value1,
-                "message": "hello"
-            },
-            {
-                timestamp_key2: timestamp_value2,
-                "message": "world"
-            }
-        ]);
-        assert_eq!(json, expected_json);
-
-        let headers = parts.headers;
-        let rfc1123date = headers.get("x-ms-date").unwrap();
-
-        let auth_expected = sink
-            .build_authorization_header_value(rfc1123date.to_str().unwrap(), body.len())
+        let (sink, _healthcheck) = config
+            .build_inner(context, mock_endpoint.into())
+            .await
             .unwrap();
 
-        let authorization = headers.get("authorization").unwrap();
-        assert_eq!(authorization.to_str().unwrap(), &auth_expected);
-
-        let log_type = headers.get("log-type").unwrap();
-        assert_eq!(log_type.to_str().unwrap(), "Vector");
-
-        let time_generated_field = headers.get("time-generated-field").unwrap();
-        let timestamp_key = log_schema().timestamp_key();
-        assert_eq!(
-            time_generated_field.to_str().unwrap(),
-            timestamp_key.unwrap().to_string().as_str()
-        );
-
-        let azure_resource_id = headers.get("x-ms-azureresourceid").unwrap();
-        assert_eq!(
-            azure_resource_id.to_str().unwrap(),
-            "97ce69d9-b4be-4241-8dbd-d265edcf06c4"
-        );
-
-        assert_eq!(
-            &parts.uri.to_string(),
-            "https://97ce69d9-b4be-4241-8dbd-d265edcf06c4.ods.opinsights.azure.com/api/logs?api-version=2016-04-01"
-        );
+        let event = Event::Log(LogEvent::from("simple message"));
+        run_and_assert_sink_compliance(sink, stream::once(ready(event)), &SINK_TAGS).await;
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏I know you used the prior value here but, shouldn't that be `HTTP_SINK_TAGS` (same for the other)?

Reply:
I believe all the new style sinks use `SINK_TAGS` instead of `HTTP_SINK_TAGS` (e.g., datadog logs sink).

Reply:
Thanks for the explanation Kyle! It sounds like we should extend `Driver` to support `endpoint` as well as `protocol` to avoid implementing this as a special case in every http-based sink.

Reply:
👍 I'm ok with making a follow-up issue for this, as it's tangential to the re-write.

Reply:
Sounds good.
https://github.com/vectordotdev/vector/issues/18197